## Reading, writing and displaying images with OpenCV

In [1]:
import cv2

In [2]:
import numpy as np 

In [3]:
input = cv2.imread('./input.jpg')
cv2.imshow('My image', input)
# 'waitKey' allows us to input information when an image window is open
# Blank: waits for anykey to be pressed before continuing. 
# Numbers (except 0): delay for how long you keep the window open (time in milliseconds)
cv2.waitKey()
# This closes all open windows 
# Failure to place this will cause your program to hang
cv2.destroyAllWindows()

### How are images stored?

In [5]:
# Import numpy
import numpy as np

In [6]:
print (input.shape)

(830, 1245, 3)


### Shape: dimensions of the image array
First two values give the 2d dimensions of the image h x w (here, 830 x 1245)

The '3' means that there are 3 other components (RGB) that make up this image.

In [7]:
print ('Height of Image:', int(input.shape[0]), 'pixels')
print ('Width of Image: ', int(input.shape[1]), 'pixels')

Height of Image: 830 pixels
Width of Image:  1245 pixels


### How do we save images we edit in OpenCV?

In [8]:
# Simply use 'imwrite' specificing the file name and the image to be saved
cv2.imwrite('output.jpg', input)
cv2.imwrite('output.png', input)

True